flink-table/flink-table-api-java/src/main/java/org/apache/flink/table/api/config/ExecutionConfigOptions.java

```diff
@@ -0,0 +1,202 @@
+			key("table.exec.resource.sink.parallelism")
+					.defaultValue(-1)
+					.withDescription("Sets sink parallelism, if it is <= 0, use " + TABLE_EXEC_RESOURCE_DEFAULT_PARALLELISM.key() + " to set sink parallelism.");
+
+	public static final ConfigOption<String> TABLE_EXEC_RESOURCE_EXTERNAL_BUFFER_MEMORY =
+			key("table.exec.resource.external-buffer-memory")
+					.defaultValue("10 mb")
+					.withDescription("Sets the external buffer memory size that is used in sort merge join and nested join and over window.");
+
+	public static final ConfigOption<String> TABLE_EXEC_RESOURCE_HASH_AGG_MEMORY =
+			key("table.exec.resource.hash-agg.memory")
+					.defaultValue("128 mb")
+					.withDescription("Sets the managed memory size of hash aggregate operator.");
+
+	public static final ConfigOption<String> TABLE_EXEC_RESOURCE_HASH_JOIN_MEMORY =
+			key("table.exec.resource.hash-join.memory")
+					.defaultValue("128 mb")
+					.withDescription("Sets the managed memory for hash join operator. It defines the lower limit.");
+
+	public static final ConfigOption<String> TABLE_EXEC_RESOURCE_SORT_MEMORY =
+			key("table.exec.resource.sort.memory")
+					.defaultValue("128 mb")
+					.withDescription("Sets the managed buffer memory size for sort.");
+
+	// ------------------------------------------------------------------------
+	//  Agg Options
+	// ------------------------------------------------------------------------
+
+	public static final ConfigOption<Integer> TABLE_EXEC_WINDOW_AGG_BUFFER_SIZE_LIMIT =
+			key("table.exec.window-agg.buffer-size-limit")
+					.defaultValue(100 * 1000)
+					.withDescription("Sets the window elements buffer size limit used in group window agg operator.");
+
+	// ------------------------------------------------------------------------
+	//  Async Lookup Options
+	// ------------------------------------------------------------------------
+	public static final ConfigOption<Integer> TABLE_EXEC_ASYNC_LOOKUP_BUFFER_CAPACITY =
+			key("table.exec.async-lookup.buffer-capacity")
+					.defaultValue(100)
+					.withDescription("The max number of async i/o operation that the async lookup join can trigger.");
+
+	public static final ConfigOption<String> TABLE_EXEC_ASYNC_LOOKUP_TIMEOUT =
+			key("table.exec.async-lookup.timeout")
+					.defaultValue("3 min")
+					.withDescription("The async timeout for the asynchronous operation to complete.");
+
+	// ------------------------------------------------------------------------
+	//  MiniBatch Options
+	// ------------------------------------------------------------------------
+
+	public static final ConfigOption<Boolean> TABLE_EXEC_MINIBATCH_ENABLED =
+			key("table.exec.mini-batch.enabled")
+					.defaultValue(false)
+					.withDescription("Specifies whether to enable MiniBatch optimization. " +
+							"MiniBatch is an optimization to buffer input records to reduce state access. " +
+							"This is disabled by default. To enable this, users should set this config to true.");
```

Comment:
add description: "NOTE: If this is true, TABLE_EXEC_MINIBATCH_ALLOW_LATENCY.key() and TABLE_EXEC_MINIBATCH_SIZE.key() must be greater than zero."